Sources/StreamChatUI/Common Views/CurrentChatUserAvatarView_Tests.swift

```diff
@@ -3,21 +3,69 @@
+    var defaultAvatarView: CurrentChatUserAvatarView! = CurrentChatUserAvatarView().withoutAutoresizingMaskConstraints
+    var customAvatarView: CustomCurrentChatUserAvatarView! = CustomCurrentChatUserAvatarView().withoutAutoresizingMaskConstraints
+    var mockedCurrentUserController: CurrentChatUserController_Mock<NoExtraData>! = .mock()
 
-    func test_default_currentChatUserAvatarView_withImage_looksAsExpected() {
-        let testImage = TestImages.vader.image
-        avatarView.accessibilityPerformMagicTap()
-        AssertSnapshot(avatarView)
+//    test_defaultAppearance() ✅
+//    test_contentIsLoaded() // klidne vic casu, treba "s" a "bez" online indicator
+//    test_customDefaultAppearance()
+//    test_customSubclassing() ✅
+
+    func test_customApperanceHook_currentChatUserAvatarView_withImage_looksAsExpected() {
+        mockedCurrentUserController.currentUser_mock = .init(
+            id: "Yoda",
+            imageURL: TestImages.yoda.url
+        )
+        defaultAvatarView.controller = mockedCurrentUserController
+
+        defaultAvatarView.applyDefaultAppearance()
+
+        AssertSnapshot(defaultAvatarView, record: true)
+    }
+
+    func test_customApperanceHook_currentChatUserAvatarView_withoutImage_looksAsExpected() {
+        AssertSnapshot(defaultAvatarView, record: true)
+    }
+
+    func test_defaultAppearance_currentChatUserAvatarView_withImage_looksAsExpected() {
+        mockedCurrentUserController.currentUser_mock = .init(
+            id: "Yoda",
+            imageURL: TestImages.yoda.url
+        )
+        defaultAvatarView.controller = mockedCurrentUserController
+        AssertSnapshot(defaultAvatarView, record: true)
     }
 
-    func test_default_currentChatUserAvatarView_withoutImage_looksAsExpected() {
-        // What is default behavior of AvatarView when no image?
-        // Shouldn't be there some default placeholder image?
-        AssertSnapshot(avatarView)
+    func test_defaultAppearance_currentChatUserAvatarView_withoutImage_looksAsExpected() {
+        AssertSnapshot(defaultAvatarView, record: true)
+    }
+
+    func test_setupAppearance_currentChatUserAvatarView_withImage_looksAsExpected() {
+        mockedCurrentUserController.currentUser_mock = .init(
+            id: "Yoda",
+            imageURL: TestImages.yoda.url
+        )
+        customAvatarView.controller = mockedCurrentUserController
+        AssertSnapshot(customAvatarView, record: true)
+    }
+
+    func test_setupAppearance_currentChatUserAvatarView_withoutImage_looksAsExpected() {
+        AssertSnapshot(customAvatarView, record: true)
+    }
+}
+
+class CustomCurrentChatUserAvatarView: CurrentChatUserAvatarView {
+    override func setUpAppearance() {
+        super.setUpAppearance()
+
+        backgroundColor = .green
+        avatarView.imageView.backgroundColor = .blue
+        avatarView.imageView.alpha = 0.2
     }
 }
```

Comment:
Ideally in the subclass, you should change something you can't change using the appearance hook, like layout for example.

Reply:
Yep you are right, Will do this in next commit.